rfc/text-field.md

```diff
@@ -0,0 +1,353 @@
+
+```
+┌──────────────────────────────────────────────────────────────────────────────┐
+│                                  TextIndex                                   │
+├──────────────────────────────────────────────────────────────────────────────┤
+│                                                                              │
+│ ┌──────────────────┐                                ┌──────────────────┐     │
+│ │   Prefix Tree    │             Postings           │   Suffix Tree    │     │
+│ └──────────────────┘                                └──────────────────┘     │
+│ │                  │                                │                  │     │
+│ │  ┌────────────┐  │          ┌────────────┐        │  ┌────────────┐  │     │
+│ │  │   apple    │──┼─────────>│ key1: [0]  │<───────┼──│   elppa    │  │     │
+│ │  └────────────┘  │          │ key2: [3]  │        │  └────────────┘  │     │
+│ │                  │          └────────────┘        │                  │     │
+│ │                  │                                │                  │     │
+│ │  ┌────────────┐  │          ┌────────────┐        │  ┌────────────┐  │     │
+│ │  │   banana   │──┼─────────>│ key1: [1]  │<───────┼──│   ananab   │  │     │
+│ │  └────────────┘  │          │ key3: [2]  │        │  └────────────┘  │     │
+│ │                  │          └────────────┘        │                  │     │
+│ │                  │                                │                  │     │
+│ │  ┌────────────┐  │          ┌────────────┐        │  ┌────────────┐  │     │
+│ │  │   cherry   │──┼─────────>│ key2: [0]  │<───────┼──│   yrrehc   │  │     │
+│ │  └────────────┘  │          │ key3: [8]  │        │  └────────────┘  │     │
+│ │                  │          └────────────┘        │                  │     │
+│ │                  │                                │                  │     │
+│ └──────────────────┘                                └──────────────────┘     │
+│                                                                              │
+└──────────────────────────────────────────────────────────────────────────────┘
+```
+
+
+### Search query operators
+
+Unlike the Vector, Tag and Numeric search operators the specification of a field is optional. If the field specification is omitted then a match is declared if any field matches the search operator.
+
+#### Term matching
+
+There are three types of term matching: exact, wildcard and fuzzy. Exact term matching is self-descriptive, i.e., only keys containing exactly the specified text are matched.
+
+Wildcard matching provides a subset of reg-ex style matching of terms.
+Initially, only a single wildcard specifier ```*``` is allowed which matches any number of characters in a term.
+The wildcard can be at any position within the term, providing prefix, suffix and infix style matching.
+Note, in this proposal, the second prefix tree is required to perform infix and suffix matching.
+
+#### Phrase matching
```

Comment:
It might be assumed already, but the different queries types includes the following. We will support all 3 types, right? 
    1. Single token: `FT.SEARCH books-idx '@title:dogs'`
    2. Phrase: `FT.SEARCH books-idx '@title:"hello world"'`
    3. Grouped: `FT.SEARCH books-idx '(@title:dogs | @title:cats)'`